mix.exs

```diff
@@ -69,13 +70,72 @@ defmodule Popcorn.MixProject do
 
   defp docs do
     [
-      main: "readme",
-      extras: ["README.md", "LICENSE"],
+      main: "introduction",
+      favicon: "assets/favicon.png",
+      logo: "assets/logo.svg",
+      extras: [
+        "pages/getting_started/introduction.md",
+        "pages/getting_started/first_steps.md",
+        "pages/guides/simple_project.md",
+        "pages/guides/runtime.md",
+        "pages/reference/architecture.md",
+        "pages/reference/limitations.md",
+        "JS Documentation": [url: "js-api/index.html"]
+      ],
+      groups_for_extras: [
+        "Getting started": ~r"/getting_started/",
+        Guides: ~r"/guides/",
+        Reference: ~r"(?:/reference/|js-api/)"
+      ],
+      groups_for_modules: [
+        "WebAssembly API": [Popcorn.Wasm, Popcorn.TrackedObject]
+      ],
       formatters: ["html"],
+      before_closing_head_tag: &before_closing_head_tag/1,
       source_ref: "v#{@version}"
     ]
   end
 
+  # Enable mermaid.js support in docs
+  defp before_closing_head_tag(:html) do
+    """
+    <script defer src="https://cdn.jsdelivr.net/npm/mermaid@11.10.1/dist/mermaid.min.js"></script>
+    <script>
+      let initialized = false;
+
+      window.addEventListener("exdoc:loaded", () => {
+        if (!initialized) {
+          mermaid.initialize({
+            startOnLoad: false,
+            theme: document.body.className.includes("dark") ? "dark" : "default"
+          });
+          initialized = true;
+        }
+
+        let id = 0;
+        for (const codeEl of document.querySelectorAll("pre code.mermaid")) {
+          const preEl = codeEl.parentElement;
+          const graphDefinition = codeEl.textContent;
+          const graphEl = document.createElement("div");
+          const graphId = "mermaid-graph-" + id++;
+          mermaid.render(graphId, graphDefinition).then(({svg, bindFunctions}) => {
+            graphEl.innerHTML = svg;
+            bindFunctions?.(graphEl);
+            preEl.insertAdjacentElement("afterend", graphEl);
+            preEl.remove();
+          });
+        }
+      });
+    </script>
+    """
+  end
+
+  defp generate_js_docs(_) do
+    Mix.shell().cmd(
+      "npx documentation build priv/static-template/wasm/popcorn.js -f html -o doc/js-api"
```

Comment:
This is what Phoenix uses. I've tested `js-doc` as well (results are here https://dev.bblaszkow.pl/popcorn_doc/js-doc/index.html), but I've decided it's way worse, at least without any customisation.

If someone is willing to handle the theming/customisation or knows a better alternative, I'm open to changing that

Reply:
I found [typedoc](https://typedoc.org). Looks a bit like apple's docs and integrates with typescript. Example: https://typedoc.org/example/classes/Customer.html

Reply:
Looks nice, the problem is we don't have typescript :stuck_out_tongue: I mean I've tested it and it complains about lack of tsconfig:
```
[warning] The entry point ./priv/static-template/wasm/popcorn.js is not referenced by the 'files' or 'include' option in your tsconfig
[error] Unable to find any entry points. See previous warnings
```

If we want to migrate to that, I'd leave it as a separate issue to migrate to TS and add typedoc